app/main.py

```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> str | int:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        if person.get("wife"):
+            male_man = Person.people[person["name"]]
```

Comment:
Try to optimise duplicated code ( Same values in variables `male_man , female_woman` ) 
DRY )